meson.build

```diff
@@ -18,15 +18,7 @@ so_version = ['0', '0', '0']
 
 add_project_arguments('-Wno-unused-parameter', language: 'c')
 add_project_arguments(
-	'-DWLR_SRC_DIR="@0@"'.format(meson.source_root()),
-	language: 'c',
-)
-add_project_arguments(
-	'-I@0@'.format(meson.build_root()),
-	language: 'c',
-)
-add_project_link_arguments(
-	'-Wl,-rpath,@0@'.format(meson.build_root()),
+	'-DWLR_SRC_DIR="@0@"'.format(meson.current_source_dir()),
```

Comment:
the breakage comes from this line (source_root to current_source_dir), I think; that'll be different as a submodule.
But frankly the problem is in glgen, trying to include the full path from build dir is insane; something like this fixes @atomnuker's problem:
```diff
diff --git a/glgen.sh b/glgen.sh
index b27750e4..2dc24eec 100755
--- a/glgen.sh
+++ b/glgen.sh
@@ -77,7 +77,7 @@ EOF
 
 cat > $OUT_C << EOF
 #include <wlr/util/log.h>
-#include "$OUT_H"
+#include "$(basename "$OUT_H")"
 $DEFN
 
 bool load_$BASE(void) {
```